### ISSUE TITLE
Developer needed for paid OSPOS modules

### ISSUE THREAD

**Issue by objecttothis** [opensourcepos/opensourcepos#2874]
### Background information

We are a non-profit bookstore and someone is wanting to donate some money to have some modules implemented.  Please let me know if you are interested.

Completed modules will need to be submitted as a PR, pass code review, be written in a way that can be used by everyone as well as be free of major flaws.  The code then becomes a part of this project and is not owned by the developer.

### New Feature

The following modules are needed.  You could take one or take several.  Let me know which you are interested in and we can negotiate the price on it.

- Creating LONGTEXT attribute type capable of handling larger blocks of text such as sample chapters, long product descriptions, etc.  It must be able to contain carriage return \n.  

- Implement precision setting to the DECIMAL attribute type.  This would be similar to the precision setting for money and quantity in the config, but implemented on a per-attribute level.  This likely involves
1. adding a new column to the attribute_definitions table to store the precision setting.
2. writing the business logic to display and store the attribute precision in the attribute view and make the setting invisible to non-decimal attribute types.
3. a default precision of 3 if the precision is not set (for those that do not wish to modify from current behavior).
4. Modifying the body of code to properly display the precision in tables (the data is stored with the full precision but just displayed with specified precision.

- Create a checkbox in the attribute definition view that makes the attribute a required or default attribute.  Default attributes are automatically added to the item/form (for new and existing items) and marked as required so that the change or new item cannot be submitted until those attributes are given acceptable values.  Storing information about whether an attribute is required/default might be able to be stored in the flags column of the attribute_definitions row.  The default should be unchecked for those who do not wish to used the feature.

- Persistent items table sort order and column visibility.  This feature would need to be customizable in the configuration.  Currently, setting which columns appear in the items table can be done, but does not persist across sessions.  Table visibility for attributes can be controlled in the attribute definition, but there is no such control for column visibility for other columns in the items table.  Implementing the same or similar flag system for each non-attribute column in the item table is what's needed here, so that when the flag is not set, that column does not appear in the items table unless added on a per session basis.  The same setting section should dictate the default sort order of results.  The default settings for visibility and sort order should be the same as current for those who do not wish to use this feature.

**Comment by jekkos** [comment]
@objecttothis weren't you already done implementing the checkbox? For reference I added the original issues #2470 #1909 #2847 #2739.

**Comment by objecttothis** [comment]
> 
> 
> @objecttothis weren't you already done implementing the checkbox? For reference I added the original issues #2470 #1909 #2847 #2739.

Yes.  None of the above enhancements are checkbox attribute types.  The above issues are all things that I am going to eventually do, but the donation is paying someone to speed them along.

**Comment by objecttothis** [comment]
I think what you might be seeing is the third issue, but that is making an attribute a default or required attribute.  It just uses a checkbox to toggle it as being default.

**Comment by objecttothis** [comment]
@opensourcepos/org anyone interested?

**Comment by jekkos** [comment]
I don't directly need any of it myself, but will let you know if I consider to take on one of these stories.

**Comment by objecttothis** [comment]
> 
> 
> I don't directly need any of it myself, but will let you know if I consider to take on one of these stories.

Thanks.  I'm starting the LONGTEXT attribute right now while I wait for people to do code reviews on #2821

**Comment by rocky8roy** [comment]
@objecttothis Hey
I can implement any feature you want
kindly contact me on rocky8roy@gmail.com to discuss further
thank you
